.github/workflows/build-main-self-hosted.yaml

```diff
@@ -73,8 +73,8 @@ jobs:
             *.args.BASE_IMAGE=${{ needs.load-env.outputs[format('{0}', matrix.base_image_env)] }}
             *.args.SETUP_ARGS=${{ matrix.setup-args }}
             *.args.LIB_DIR=${{ matrix.lib_dir }}
-            *.cache-from=type=registry,ref=ghcr.io/${{ github.repository }}:buildcache-${{ matrix.name }}
-            *.cache-to=type=registry,ref=ghcr.io/${{ github.repository }}:buildcache-${{ matrix.name }},mode=max
+            *.cache-from=type=registry,ref=ghcr.io/${{ github.repository }}:buildcache-${{ matrix.name }}-${{ matrix.lib_dir }}
+            *.cache-to=type=registry,ref=ghcr.io/${{ github.repository }}:buildcache-${{ matrix.name }}-${{ matrix.lib_dir }},mode=max
           tag-suffix: ${{ matrix.additional-tag-suffix }}-arm64
```

Comment:
Should we keep the suffix for the generate image it's cache consistent? 

At the moment, it uses `-arm64` for the actual image, but it seems this generates build-cache with `--aarch64`
![image](https://github.com/autowarefoundation/autoware/assets/43976834/2d348e9c-8fe1-4e42-a629-20d8f927e589)
![image](https://github.com/autowarefoundation/autoware/assets/43976834/ad2ce629-a8a1-4886-a514-a6cddbeec6ca)

We should either use `-${{ matrix.lib_dir }}` for both or `-arm64` for both.

Reply:
Same goes applies to amd64 and x64_64

Reply:
I've changed to `-amd64`, `-arm64` suffixes. https://github.com/autowarefoundation/autoware/pull/4826/commits/29e5193f660312946a1a252e63bfcb2039bedeeb